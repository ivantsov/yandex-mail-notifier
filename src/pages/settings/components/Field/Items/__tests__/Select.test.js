import React from 'react';
import renderer from 'react-test-renderer';
import Select from '../Select';

function testCase(props) {
    const component = renderer.create(
        <Select
            name="field"
            onChange={jest.fn()}
            {...props}
        />,
    );

    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();

    return tree;
}

describe('settings/Select', () => {
    it('no options', () => {
        testCase({
            value: 123,
            options: [],
        });
    });

    describe('with options', () => {
        function testCaseWithOptions({
            value,
            newValue,
            options,
        }) {
            const name = 'field';
            const onChange = jest.fn();

            const tree = testCase({
                name,
                value,
                options,
                onChange,
            });

            tree.props.onChange({target: {value: newValue}});
            expect(onChange).lastCalledWith(name, newValue);
        }

        it('number options', () => {
            testCaseWithOptions({
                value: 1,
                newValue: 10,
                options: [{
                    label: 'option1',
                    value: 1,
                }, {
                    label: 'option2',
                    value: 2,
                }],
            });
        });

        it('string options', () => {
            testCaseWithOptions({
                value: 'opt1',
                newValue: 'opt2',
                options: [{
                    label: 'option1',
                    value: 'opt1',
                }, {
                    label: 'option2',
                    value: 'opt2',
                }],
            });
        });
    });
});
