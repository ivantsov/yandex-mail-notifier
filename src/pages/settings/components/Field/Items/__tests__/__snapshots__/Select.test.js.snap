// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`settings/Select no options 1`] = `
<select
  id="field"
  name="field"
  onChange={[Function]}
  value={123}
/>
`;

exports[`settings/Select with options number options with zero value 1`] = `
<select
  id="field"
  name="field"
  onChange={[Function]}
  value={1}
>
  <option
    value={0}
  >
    option0
  </option>
  <option
    value={1}
  >
    option1
  </option>
</select>
`;

exports[`settings/Select with options number options without zero value 1`] = `
<select
  id="field"
  name="field"
  onChange={[Function]}
  value={1}
>
  <option
    value={1}
  >
    option1
  </option>
  <option
    value={2}
  >
    option2
  </option>
</select>
`;

exports[`settings/Select with options string options 1`] = `
<select
  id="field"
  name="field"
  onChange={[Function]}
  value="opt1"
>
  <option
    value="opt1"
  >
    option1
  </option>
  <option
    value="opt2"
  >
    option2
  </option>
</select>
`;
